strawberryfields/program.py

```diff
@@ -203,6 +203,66 @@ def __len__(self):
         """
         return len(self.circuit)
 
+    def __eq__(self, prog):
+        """Equality operator for programs.
+
+        .. note::
+
+            This operation does not account for logically equivalent programs where the order
+            of operations or modes do not matter. It simply checks that the operations and
+            parameters are identically applied. For a more thorough check, use the
+            ``Program.equivalence()`` method instead.
```

Comment:
Does this show in the documentation? 🤔 It might have to be added elsewhere, like in the main `Program` docstring.

Reply:
Probably not, no. Moved to `Program` class docstring.